### PR TITLE
Fix for mqttmessage_getTopicLevels if topic name is empty

### PR DESCRIPTION
**Pull request by ewertons**

### CODE REVIEW

**Comment by massand** [comment]
>         }

do this below instead of just const_topic_name

---
Refers to: src/mqtt_message.c:211 in a4bd1de. [](commit_id = a4bd1de8f6bfc3dd94fa1b8531bdc788e0d64d69, deletion_comment = False)

**Comment by massand** [comment]
>         if (handle->topicName == NULL)

do this below instead of just const_topic_name, please :)

---
Refers to: src/mqtt_message.c:204 in a4bd1de. [](commit_id = a4bd1de8f6bfc3dd94fa1b8531bdc788e0d64d69, deletion_comment = False)

File: src/mqtt_message.c [hunk 1]
```diff
@@ -226,22 +226,31 @@ int mqttmessage_getTopicLevels(MQTT_MESSAGE_HANDLE handle, char*** levels, size_
...
     {
         MQTT_MESSAGE* msgInfo = (MQTT_MESSAGE*)handle;
-        const char* delimiters[1];
 
-        delimiters[0] = "/";
-
-        // Codes_SRS_MQTTMESSAGE_09_002: [ The topic name, excluding the property bag, shall be split into individual tokens using "/" as separator ]
-        // Codes_SRS_MQTTMESSAGE_09_004: [ The split tokens shall be stored in `levels` and its count in `count` ]
-        if (StringToken_Split(msgInfo->topicName, strlen(msgInfo->topicName), delimiters, 1, false, levels, count) != 0)
+        if (msgInfo->const_topic_name == NULL)
```
Comment (jebrando) [inline src/mqtt_message.c]: >msgInfo->const_topic_name [](start = 12, length = 25)

The const_topic_name is for when the caller is allocating the topic, handle->topic_name is used when the caller wants umqtt to allocate the topic.  You'll should set up something like this:

const char* use_topic_name = msgInfo->topic_name;
if (use_topic_name == NULL)
{
    use_topic_name = msgInfo->const_topic_name;
}

if (use_topic_name == NULL)
{
    LogError("Topic name is NULL")
    ...

Comment (ewertons) [reply src/mqtt_message.c]: Addressed. Thanks

**Pull request merged**